src/Microsoft.AspNetCore.Proxy/ProxyAdvancedExtensions.cs

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Net.WebSockets;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Proxy
+{
+    public static class ProxyAdvancedExtentions
+    {
+        private static readonly string[] NotForwardedWebSocketHeaders = new[] { "Connection", "Host", "Upgrade", "Sec-WebSocket-Key", "Sec-WebSocket-Version" };
+        private const int DefaultWebSocketBufferSize = 4096;
+
+        public static Uri ToWebSocketScheme(this Uri uri)
```

Comment:
Please, no. I really, really need those methods (maybe except the ToWebSocketScheme for advanced scenarios addressing for example #35.

They are not visible unless you add use Microsoft.AspNetCore.Proxy; All other classes are in dependency injection related namespaces. And those are intended for advanced use scanarios as I had described. I really put a big effort to design those methods naming in a usable way.